collect_app/src/test/java/org/odk/collect/android/backgroundwork/AutoSendTaskSpecTest.kt

```diff
@@ -64,4 +66,9 @@ class AutoSendTaskSpecTest {
         AutoSendTaskSpec().getTask(ApplicationProvider.getApplicationContext(), inputData).get()
         verify(instanceAutoSender).autoSendInstances(projectId)
     }
+
+    @Test
+    fun `numberOfRetries should not be limited`() {
+        assertThat(AutoSendTaskSpec().numberOfRetries, `is`(-1))
```

Comment:
I renamed `numberOfRetries` to `maxRetries` but I'm not sure if I understand the fist part correctly. You added this comment ins a commit. The final version doe not have that raw -1 but it's represented by a const `UNLIMITED_NUMBER_OF_RETRIES`. Is there anything else you would like to change here?

Reply:
Sorry maybe that ended up being unclear. What I mean is using an optional type (`Int?`) instead of a non-optional and then having `null` denote "no retries". Then the check can be `if (spec.maxRetries == null...` and there'd be no need for a special constant. I know using `-1` is pretty common in Java, but I feel like having safe nullables in Kotlin gives us a clearer way of dealing with that kind of "optional" value.

Reply:
Ah I see but making `maxRetries` nullable would make using it a bit more difficult even in kotlin for example in `WorkerAdapter` instead of:
```
this[TaskSpec.DATA_LAST_UNIQUE_EXECUTION] = (runAttemptCount >= spec.maxRetries).toString()
```
I would need to check if `maxRetries` is not null first (and it's not the only place like that).
Of course I could use this for example:
```
this[TaskSpec.DATA_LAST_UNIQUE_EXECUTION] = (spec.maxRetries?.let { runAttemptCount >= it }).toString()
```
but it makes the code less readable to me.

Reply:
Yeah, I mean I think this is getting into personal preference, but I prefer the latter to the former there. The first example relies on knowledge that a negative number is being used to denote "no retries" whereas I'd argue that the second example makes it clearer that there is no last unique execution if there is no `maxRetries`.  The first example also relies on a bit of a "trick": a negative number is always lower than the run attempt count which is concise, but I don't personally feel it's clearer.

The other problem (and probably the more important one) I have with using negative numbers, is that it brings in more risk of writing buggy code. For instance, this code works with any negative value, but it'd be easy to write code that relies on `-1` specifically and have a value like `-2` cause problems. 

I think if you feel strongly that the negative number is better, then let's keep it - it's just stylistic peeve of mine.

Reply:
Ok we can do that like that, done.